lang/cs/Org.Apache.REEF.IMRU/API/UpdateResult.cs

```diff
@@ -50,22 +50,34 @@ internal bool IsNotDone
             get { return !_done; }
         }
 
-        internal bool HasMapInput
+        /// <summary>
+        /// Indicates if the UpdateResult has map input data
+        /// </summary>
+        public bool HasMapInput
```

Comment:
The question here is not what is the problem in exposing but the argument is that it is not needed to expose. BroadcastSenderReduceReceiverUpdateFunction you are mentioning is an example code and is not a good example of typical settings in which IMRU update and map functions will be written...Since Update function created UpdateResult using publicly exposed APIs in any case it has all the information to update state task in any case.. So there is no need to go via UpdateResult APIs.

So I will still be insistent on keeping the calls internal.